src/granite/transactions.cr

```diff
@@ -196,30 +181,19 @@ module Granite::Transactions
       return false
     end
     true
-  {% end %}
   end
 
   def save!
     save || raise Granite::RecordNotSaved.new(self.class.name, self)
   end
 
-  def update(**args)
-    update(args.to_h)
-  end
-
-  def update(args : Granite::ModelArgs)
-    set_attributes(args.transform_keys(&.to_s))
-
+  def update(*args, **named_args)
+    initialize(*args, **named_args)
```

Comment:
Is it ok to call `initialize`?

Reply:
🤷‍♂ probably?

Reply:
It might be better to put the functionality in a shared method elsewhere, even if it means repeating the code so that crystal still sees the property initialization correctly. If the idea here is to encourage developers to create their own init methods, we should avoid calling it in hidden places. Developers are likely to attach weird logic there.

Reply:
Maybe some private initializer that's only used internally?

Reply:
Yeah that's what I mean by duplicating the code.

Reply:
If we go the "remove default initializer route", which seems to be the popular opinion, then this wouldn't matter anymore.  

We should be able to support `#create` as it's essentially the same flow as calling `#new` but also calls `#save` before returning the object.  Supporting `#update` will be harder.